src/lib/components/textarea/TextArea.component.tsx

```diff
@@ -2,22 +2,42 @@ import React, { forwardRef, HTMLProps } from 'react';
 import styled, { css } from 'styled-components';
 import { spacing } from '../../spacing';
 import { getTheme } from '../../utils';
-type Props = Omit<HTMLProps<HTMLTextAreaElement>, 'as'>;
+
+type TextAreaVariant = 'code' | 'text';
+type Props =
+  | Omit<HTMLProps<HTMLTextAreaElement>, 'as'> & { variant?: TextAreaVariant };
 type RefType = HTMLTextAreaElement | null;
 
-const TextAreaContainer = styled.textarea`
+const TextAreaContainer = styled.textarea<{
+  variant: TextAreaVariant;
+  width?: string | number;
```

Comment:
Just thinking maybe we can use `CSSProperties` which has already typed for us ? 
```suggestion
  width?: CSSProperties['width'];
```

Reply:
Indeed thanks for pointing this, much better !